frontend/src/pages/dashboard/components/orders/orders.tsx

```diff
@@ -1,57 +1,147 @@
-  useAppMap,
   useAppSelector,
+  useCallback,
   useEffect,
-  useRef,
+  useMemo,
+  useQueryParameters,
   useState,
 } from '~/libs/hooks/hooks.js';
+import {
+  DEFAULT_PAGE_INDEX,
+  DEFAULT_PAGE_SIZE,
+} from '~/libs/hooks/use-app-table/libs/constant.js';
 import { type PlaceLatLng } from '~/libs/packages/map/libs/types/types.js';
-import { actions as ordersActions } from '~/slices/orders/orders.js';
-import { selectOrders } from '~/slices/orders/selectors.js';
+import {
+  type Coordinates,
+  type OrderResponseDto,
+  type OrderStatusValues,
+} from '~/libs/types/types.js';
+import { actions as ordersActions } from '~/slices/orders/order.js';
+import { selectOrdersBusiness } from '~/slices/orders/selectors.js';
 
+import { MapOrders } from './map-orders.js';
 import styles from './styles.module.scss';
 
 const Orders: React.FC = () => {
   const dispatch = useAppDispatch();
 
-  const orders = useAppSelector(selectOrders);
+  const { orders, total, dataStatus } = useAppSelector(selectOrdersBusiness);
 
-  const [endPointMarkers, setEndPointMarkers] = useState<
-    google.maps.LatLngLiteral[]
-  >([]);
+  const { setQueryParameters, removeQueryParameters, searchParameters } =
+    useQueryParameters();
+
+  const [endPointMarkers, setEndPointMarkers] = useState<Coordinates[]>([]);
 
   const [shownRoute, setShownRoute] = useState<PlaceLatLng>();
 
-  const mapReference = useRef<HTMLDivElement>(null);
+  const [filter, setFilter] = useState<{
+    status: OrderStatusValues | null;
+  }>({
+    status: null,
+  });
+
+  const [pageIndex, setPageIndex] = useState<number>(DEFAULT_PAGE_INDEX);
 
   useEffect(() => {
-    void dispatch(ordersActions.getBusinessOrders());
-  }, [dispatch]);
+    setQueryParameters({
+      size: DEFAULT_PAGE_SIZE,
+      page: pageIndex,
+    });
+  }, [pageIndex, setQueryParameters]);
+
+  useEffect(() => {
+    if (filter.status) {
+      setQueryParameters({
+        size: DEFAULT_PAGE_SIZE,
+        page: DEFAULT_PAGE_INDEX,
+        status: filter.status,
+      });
+    } else {
+      removeQueryParameters('status');
+    }
+  }, [filter.status, removeQueryParameters, setQueryParameters]);
+
+  useEffect(() => {
+    void dispatch(ordersActions.getBusinessOrders(searchParameters.toString()));
+  }, [dispatch, searchParameters]);
 
   useEffect(() => {
     setEndPointMarkers(orders.map((order) => order.endPoint));
   }, [orders]);
 
-  useAppMap({
-    mapReference: mapReference,
-    points: endPointMarkers,
-    shownRoute,
-    center: null,
-    destination: null,
-  });
+  const totalPages = useMemo(
+    () => Math.ceil(total / DEFAULT_PAGE_SIZE),
+    [total],
+  );
+
+  const handleChangePage = useCallback((pageIndex: number) => {
+    setPageIndex(pageIndex);
+  }, []);
+
+  const handleChangeFilter = useCallback(
+    (filter: { status: OrderStatusValues | null }) => {
+      setFilter(filter);
+      setPageIndex(DEFAULT_PAGE_INDEX);
+    },
+    [],
+  );
+
+  const sortOrders = useMemo(
```

Comment:
why do we need sortOrders in the component? can we get already sorted data from the be?

Reply:
yes, we get them already sorted. it was double sorting